visibility-sensor.js

```diff
@@ -303,13 +306,14 @@ module.exports = createReactClass({
         visibilityRect: visibilityRect
       };
       this.setState(state);
-      this.props.onChange(isVisible, visibilityRect);
+      if (this.props.onChange) this.props.onChange(isVisible, visibilityRect);
     }
 
     return state;
   },
 
   render: function () {
+    if (this.props.children instanceof Function) return this.props.children(this.state);
```

Comment:
I would destructure `this.state` here to return explicit values. Prevents leaking of internal state details to the render callback in the future.